crates/config/src/rule/range.rs

```diff
@@ -0,0 +1,161 @@
+use ast_grep_core::{meta_var::MetaVarEnv, Doc, Language, Node, Position};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+/// Represents a position in a document
```

Comment:
NIT: the comment here needs to explain the position is in zero-based, character-wise